graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -77,8 +77,21 @@
 - Names may not conflict with each other. For instance, you may not rename both "Foo" and "Bar" to
   "Baz". You also may not rename anything to "Baz" if a type "Baz" already exists and is not also
   being renamed or suppressed.
+- Special rules apply to the renamings argument if it's iterable (e.g. if renamings is a dict).
+  Specifically, if renamings is iterable, then renamings may not contain no-op entries. In other
+  words, a string type_name may be in renamings only if there exists a type in the schema named
+  type_name (since otherwise that entry would not affect any type in the schema). Also, if type_name
+  is in renamings, then renamings[type_name] != type_name (since applying the renaming would not
+  change the type named type_name).
+  Note that these rules apply only if renamings is iterable. If renamings is a non-iterable
+  object that just implements a get() method, there's no way to enforce these rules since there's
+  no way to keep arbitrary code from doing valid but inadvisable things. These rules exist
+  specifically for iterables because that's the most common usage for rename_schema.
 """
+# Note that importing from collections.abc instead of collections is due to a pylint issue that goes
```

Comment:
Note: black auto-formatted this comment to put it above all the imports and not just the relevant import